util/git/client.go

```diff
@@ -710,12 +709,24 @@ func (m *nativeGitClient) resolveSemverRevision(revision string, refs []*plumbin
 		tag := ref.Name().Short()
 		version, err := semver.NewVersion(tag)
 		if err != nil {
-			if errors.Is(err, semver.ErrInvalidSemVer) {
-				log.Debugf("Invalid semantic version: %s", tag)
-				continue
+			switch {
```

Comment:
Instead of a switch with custom error strings, why not just use the returned error itself? i.e. only use what's currently coded as the `default` case.